### PR TITLE
[MODCON-57] Implement Delete endpoint for publication

### PR DESCRIPTION
**Pull request by bekhzodkurbonboev**
## Purpose
Resolves: [MODCON-57](https://issues.folio.org/browse/MODCON-57) Implement Delete endpoint for publication

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -174,6 +174,16 @@
           ],
           "modulePermissions": []
         },
+        {
+          "methods": [
+            "DELETE"
+          ],
+          "pathPattern": "/consortia/{consortiumId}/publications/{publicationId}/results",
```
Comment (grabsefx) [inline descriptors/ModuleDescriptor-template.json:181]: shouldn't it be '/consortia/{consortiumId}/publications/{publicationId} ?'

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -326,7 +336,8 @@
         "consortia.sharing-instances.collection.get",
         "consortia.publications.item.post",
         "consortia.publications.item.get",
-        "consortia.publications-results.item.get"
+        "consortia.publications-results.item.get",
+        "consortia.publications-results.item.delete"
```
Comment (grabsefx) [inline descriptors/ModuleDescriptor-template.json:340]: should be `consortia.publications.item.delete`

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java [hunk 3]
```diff
@@ -293,4 +293,18 @@ public PublicationResultCollection getPublicationResults(UUID consortiumId, UUID
       .totalRecords(ptrEntities.getSize());
   }
 
+  @Override
```
Comment (grabsefx) [inline src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java:296]: requires `@Transactional` annotation in order to delete records from two tables transactionally

File: src/main/resources/swagger.api/publications.yaml [hunk 4]
```diff
@@ -53,6 +53,19 @@ paths:
           $ref: '#/components/responses/BadRequest'
         '500':
           $ref: '#/components/responses/InternalServerError'
+    delete:
+      summary: Delete publication result by Id
+      operationId: deletePublicationResultById
```
Comment (grabsefx) [inline src/main/resources/swagger.api/publications.yaml:58]: Operation name `deletePublicationById` would be more relevant
summary value also needs to be updated accordingly

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -374,6 +385,11 @@
       "displayName": "publish coordinator results",
       "description": "publish coordinator results"
     },
+    {
+      "permissionName": "consortia.publications-results.item.delete",
+      "displayName": "delete publish coordinator results",
```
Comment (grabsefx) [inline descriptors/ModuleDescriptor-template.json:390]: `delete publish coordinator records` or `delete publication`

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (bekhzodkurbonboev) [reply descriptors/ModuleDescriptor-template.json]: Thank you, corrected.

File: descriptors/ModuleDescriptor-template.json (hunk 2 above)
Comment (bekhzodkurbonboev) [reply descriptors/ModuleDescriptor-template.json]: Fixed.

File: src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java (hunk 3 above)
Comment (bekhzodkurbonboev) [reply src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java:297]: Thank you, added.

File: src/main/resources/swagger.api/publications.yaml (hunk 4 above)
Comment (bekhzodkurbonboev) [reply src/main/resources/swagger.api/publications.yaml]: Fixed.

File: descriptors/ModuleDescriptor-template.json (hunk 5 above)
Comment (bekhzodkurbonboev) [reply descriptors/ModuleDescriptor-template.json]: Refactored.

**Review by SerhiiNosko — APPROVED** [review]

**Review by SinghAdes — APPROVED** [review]

**Pull request merged**